privval/grpc/client_test.go

```diff
@@ -33,11 +33,7 @@ func dialer(pv types.PrivValidator, logger log.Logger) (*grpc.Server, func(conte
 
 	privvalproto.RegisterPrivValidatorAPIServer(server, s)
 
-	go func() {
-		if err := server.Serve(listener); err != nil {
-			panic(err)
-		}
-	}()
+	go func() { require.NoError(t, server.Serve(listener)) }()
```

Comment:
Not sure this needs the closing 'func' invocation?

Reply:
:facepalm:

Reply:
Ah, I mean that I believe you can just write `go require.NoError(t, server.Serve(listener))`, no?

Reply:
No, because `go`, like `defer`, evaluates all the expression arguments (the function to call, the arguments) up to the point of the call, and only does the call after the jump.